tv_apps/smart-home/js/home.js

```diff
@@ -187,12 +193,77 @@
           cardManager: that.cardManager
         });
         that._cardPicker.on('hide', that.onCardPickerHide.bind(that));
+        that._cardPicker.on('show', that.onCardPickerShow.bind(that));
+      });
+    },
+
+    initContextMenu: function () {
+      this.moveMenuItem.addEventListener('click', () => {
+          this.edit.toggleEditMode();
+          this.edit.once('arranged', () => this.edit.toggleEditMode());
+          this.edit.toggleArrangeMode();
+      });
+
+      this.renameMenuItem.addEventListener('click', () => {
+        var focus = this.spatialNavigator.getFocusedElement();
+        this.edit.toggleEditMode();
+        this.edit.once('rename-card-ended', () => this.edit.toggleEditMode());
+        this.edit.renameCard(focus, focus.getNodeFromItem(focus.currentItem));
+      });
+
+      this.removeMenuItem.addEventListener('click', () => {
+        var focus = this.spatialNavigator.getFocusedElement();
+        this.edit.toggleEditMode();
+        this.edit.once('delete-card-ended', () => this.edit.toggleEditMode());
+        this.edit.openDeleteCardDialog(focus,
+          focus.getNodeFromItem(focus.currentItem));
+      });
```

Comment:
We have to toggle off the edit mode after deleting-card job ends.
Because the open of dialog is non-blocking so if we do the following code:
```
this.edit.toggleEditMode();
this.edit.openDeleteCardDialog(……..);
this.edit.toggleEditMode();
```
2 issues could happen:
(1) We would exit the edit mode immediately while the deleting-card dialog still remains opened.
(2) If we exit the edit mode before deleting card transition finishes, during exiting the edit mode, folder would still be considered not empty so empty folder would not be removed after exiting the edit mode